### PR TITLE
add 1.3.8-release doc

### PR DESCRIPTION
**Pull request by dailidong**
add 1.3.8-release doc

### CODE REVIEW

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+1. This document is intended for users with a basic understanding of Ambari
+2. This document is a description of adding the DolphinScheduler service to the installed Ambari service
+3. This document is based on version 2.5.2 of Ambari 
+
+#### Installation preparation
+
+1. Prepare the RPM packages
+
+   - It is generated by executing the command ```mvn -U clean install -Prpmbuild -Dmaven.test.skip=true -X``` in the project root directory (In the directory: dolphinscheduler-dist/target/rpm/apache-dolphinscheduler/RPMS/noarch )
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:13]:
```suggestion
   - It is generated by executing the command `mvn -U clean install -Prpmbuild -Dmaven.test.skip=true -X` in the project root directory (In the directory: dolphinscheduler-dist/target/rpm/apache-dolphinscheduler/RPMS/noarch)
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+2. Create an installation for DS，who have read and write access to the installation directory (/opt/soft)
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:15]:
```suggestion
2. Create an installation for DolphinScheduler with the user have read and write access to the installation directory (/opt/soft)
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+3. Install with rpm package
+
+   - Manual installation (recommended）：
+      - Copy the prepared RPM packages to each node of the cluster.
+      - Execute with DS installation user: ```rpm -ivh apache-dolphinscheduler-xxx.noarch.rpm```
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:21]:
```suggestion
      - Execute with DolphinScheduler installation user: `rpm -ivh apache-dolphinscheduler-xxx.noarch.rpm`
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      - Mysql-connector-java packaged using the default POM file will not be included.
+      - The RPM package was packaged in the project with the installation path of /opt/soft. 
+        If you use mysql as the database, you need add it manually.
+      
+   - Automatic installation with ambari
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:26]:
```suggestion
   - Automatic installation with Ambari
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      - Copy the prepared RPM packages to each node local yum source
+
+4. Copy plug-in directory
+
+   - copy directory ambari_plugin/common-services/DOLPHIN to ambari-server/resources/common-services/
+   - copy directory ambari_plugin/statcks/DOLPHIN to ambari-server/resources/stacks/HDP/2.6/services/--stack version is selected based on the actual situation
+
+5. Initializes the database information
+
+   ```
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:37]:
```suggestion
   ```sql
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+   -- Create the database for the DolphinScheduler：dolphinscheduler
+   CREATE DATABASE dolphinscheduler DEFAULT CHARACTER SET utf8 DEFAULT COLLATE
+   utf8_general_ci;
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:40]:
```suggestion
   CREATE DATABASE dolphinscheduler DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_general_ci;
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+   
+   -- Initialize the user and password for the dolphinscheduler database and assign permissions
+   -- Replace the {user} in the SQL statement below with the user of the dolphinscheduler database
+   GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'%' IDENTIFIED BY '{password}';
+   GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'localhost' IDENTIFIED BY
+   '{password}';
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:46]:
```suggestion
   GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'localhost' IDENTIFIED BY '{password}';
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+   flush privileges;
+   ```
+
+ 
+
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:51]:
```suggestion
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+#### Ambari Install DolphinScheduler
+- **NOTE: You have to install zookeeper first**
+
+1. Install DolphinScheduler on ambari web interface
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:55]:
```suggestion
1. Install DolphinScheduler on Ambari web interface
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+   GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'%' IDENTIFIED BY '{password}';
+   GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'localhost' IDENTIFIED BY
+   '{password}';
+   flush privileges;
+   ```
+
+ 
+
+#### Ambari Install DolphinScheduler
+- **NOTE: You have to install zookeeper first**
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:53]:
```suggestion
- **NOTE: You have to install Zookeeper first**
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+   ![](https://dolphinscheduler.apache.org/img/ambari-plugin/DS2_AMBARI_009.png)
+   
+   
+
+------
+
+
+
+#### Add components to the node through Ambari -- for example, add a DS Worker
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:99]:
```suggestion
#### Add components to the node through Ambari -- for example, add a DolphinScheduler Worker
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+***NOTE***: DS Logger is the installation dependent component of DS Worker in Dolphin's Ambari installation (need to add installation first; Prevent the Job log on the corresponding Worker from being checked)
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:101]:
```suggestion
***NOTE***: DolphinScheduler Logger is the installation dependent component of DS Worker in Dolphin's Ambari installation (need to add installation first; Prevent the Job log on the corresponding Worker from being checked)
```

File: docs/en-us/1.3.8/user_doc/ambari-integration.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+2. Add components -- the drop-down list is all addable
+
+   ![DS2_AMBARI_012](https://dolphinscheduler.apache.org/img/ambari-plugin/DS2_AMBARI_012.png)
+
+3. Confirm component addition
+
+   ![DS2_AMBARI_013](https://dolphinscheduler.apache.org/img/ambari-plugin/DS2_AMBARI_013.png)
+
+4. After adding DS Worker and DS Logger components
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/ambari-integration.md:115]:
```suggestion
4. After adding DolphinScheduler Worker and DolphinScheduler Logger components
```

File: docs/en-us/1.3.8/user_doc/architecture-design.md [hunk 14]
```diff
@@ -0,0 +1,332 @@
...
+  </p>
+</p>
+
+**Process definition**：Visualization formed by dragging task nodes and establishing task node associations**DAG**
+
+**Process instance**：The process instance is the instantiation of the process definition, which can be generated by manual start or scheduled scheduling. Each time the process definition runs, a process instance is generated
+
+**Task instance**：The task instance is the instantiation of the task node in the process definition, which identifies the specific task execution status
+
+**Task type**： Currently supports SHELL, SQL, SUB_PROCESS (sub-process), PROCEDURE, MR, SPARK, PYTHON, DEPENDENT (depends), and plans to support dynamic plug-in expansion, note: 其中子 **SUB_PROCESS**  It is also a separate process definition that can be started and executed separately
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/architecture-design.md:20]:
```suggestion
**Task type**： Currently supports SHELL, SQL, SUB_PROCESS (sub-process), PROCEDURE, MR, SPARK, PYTHON, DEPENDENT (depends), and plans to support dynamic plug-in expansion, note: **SUB_PROCESS**  It is also a separate process definition that can be started and executed separately
```

File: docs/en-us/1.3.8/user_doc/architecture-design.md [hunk 15]
```diff
@@ -0,0 +1,332 @@
...
+- In fact, truly decentralized distributed systems are rare. Instead, dynamic centralized distributed systems are constantly pouring out. Under this architecture, the managers in the cluster are dynamically selected, rather than preset, and when the cluster fails, the nodes of the cluster will automatically hold "meetings" to elect new "managers" To preside over the work. The most typical case is Etcd implemented by ZooKeeper and Go language.
+
+
+
+- The decentralization of DolphinScheduler is that the Master/Worker is registered in Zookeeper, and the Master cluster and Worker cluster are centerless, and the Zookeeper distributed lock is used to elect one of the Master or Worker as the "manager" to perform the task.
+
+#####  Two、Distributed lock practice
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/architecture-design.md:141]:
```suggestion
##### Two、Distributed lock practice
```

File: docs/en-us/1.3.8/user_doc/architecture-design.md [hunk 16]
```diff
@@ -0,0 +1,332 @@
...
+        // thread name： taskThreadName-processDefineId_processInstanceId_taskInstanceId
+        String threadName = event.getThreadName();
+        String[] threadNameArr = threadName.split("-");
+        // logId = processDefineId_processInstanceId_taskInstanceId
+        String logId = threadNameArr[1];
+        ...
+        super.subAppend(event);
+    }
+}
+ ```
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/architecture-design.md:288]:
```suggestion
```
```

File: docs/en-us/1.3.8/user_doc/architecture-design.md [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+
+    @Override
+    public FilterReply decide(ILoggingEvent event) {
+        if (event.getThreadName().startsWith("TaskLogInfo-")){
+            return FilterReply.ACCEPT;
+        }
+        return FilterReply.DENY;
+    }
+}
+ ```
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/architecture-design.md:311]:
```suggestion
```
```

File: docs/en-us/1.3.8/user_doc/architecture-design.md [hunk 18]
```diff
@@ -0,0 +1,332 @@
...
+- dolphinscheduler-dao provides operations such as database access.
+
+- dolphinscheduler-remote client and server based on netty
+
+- dolphinscheduler-server MasterServer and WorkerServer services
+
+- dolphinscheduler-service service module, including Quartz, Zookeeper, log client access service, easy to call server module and api module
+
+- dolphinscheduler-ui front-end module
+### Sum up
```
Comment (zhongjiajie) [inline docs/en-us/1.3.8/user_doc/architecture-design.md:329]:
```suggestion

### Sum up
```

**Review by zhongjiajie — COMMENTED** [review]
Add some minor suggest, I not sure should I direct add suggest here or submit PR for it

**Comment by dailidong** [comment]
> Add some minor suggest, I not sure should I direct add suggest here or submit PR for it

Wow, so many changes, I think you can submit a new PR next time. ^_^

**Comment by zhongjiajie** [comment]
Ok, I got it

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**